MachineLearningPrototype/dataCollection/collect_data.py

```diff
@@ -0,0 +1,169 @@
+	issue_details['url'] = issue['html_url']
+	issue_details['repository_url'] = issue['repository_url']
+	issue_details['number'] = issue_number
+	issue_details['state'] = issue['state']
+	issue_details['title'] = issue['title']
+	issue_details['created_at'] = issue['created_at']
+	issue_details['updated_at'] = issue['updated_at']
+	issue_details['issue_content_path'] = issue_content_path
+	# issue_details['test_output_path'] = test_output_path
+
+	#Store issue content at issue_content_path
+	store_on_fs(issue['body'], issue_content_path)
+	
+	#Store issue details in db.Issues Table
+	store_in_db(issue_details, "Issues", db, "url")
+
+	#To do: Add issue_details['test_output_path'] and store test_output in filesystem
+
+
+def fetch_new_issues(repo, since, auth_token, db):
+	"""
+	Fetch all new open issues in the repo since `since`
+	"""
+	num_pages = get_num_pages(repo)
+	pprint(f'Need to query {num_pages} pages')
+
+	page_number = 1
+	num_open_issues = 0
+	
+	query_url = f'https://api.github.com/repos/{repo}/issues'
+	params = {'accept': 'application/vnd.github.v3+json',
+			  'state': 'open',
+			  'since': since,
+			  'per_page': 100,
+			  'page': page_number,}
+
+	headers = None
+	if auth_token:
+		headers = {'Authorization': f'token {auth_token}'}
+
+	while page_number <= num_pages:
+		params['page'] = page_number
+		response = requests.get(query_url, headers=headers, params=params).json()
+		for r in response:
+			if 'pull_request' not in r:
+				store_issue_details(r, repo, db)	
+				num_open_issues += 1
+		page_number += 1
+	
+	pprint(f'Found {num_open_issues} open issues')
+	return num_open_issues
+
+def fetch_github_issues(args, db):
+	"""
+	Track all new open issues in the repo with the given labels
+	"""
+	repos = args.github_repo
+	auth_token = args.github_token
+	wait_time = args.github_freq * 60
+
+	while True:
+		start = time.time()
+		
+		for repo in repos:
+
+			#Get the last updated time for the repo
+			since_info = get_collection_record({'repository': repo}, None, 'LastUpdatedInfo', db)
+			if since_info:
+				since = since_info['last_updated_time']
+			else:
+				since = None
+	
+			pprint(f"Fetching open issues for {repo}")
+			num_issues = fetch_new_issues(repo, since, auth_token, db)
+
+			#Store the last updated time for the repo
+			since = datetime.now().isoformat()
+			since_record_to_store = {
+				'repository': repo, 
+				'last_updated_time': since,
+				'total_issues_collected': num_issues
+			}
+			store_in_db(since_record_to_store, "LastUpdatedInfo", db, primary_key='repository')
+			
+		end = time.time()
+		pprint(f'Time taken to fetch issues: {end-start} seconds')
+
+		time.sleep(wait_time)
+
+def main():
+	parser = argparse.ArgumentParser(description="Data collection for DeepAQAtik")
+
+	parser.add_argument('--github_token', type=str, default=None, help='GitHub Auth token')
+	parser.add_argument('--github_repo', type=str, default=['adoptium/aqa-tests'], nargs='+', help='Github repo to track')
```

Comment:
In the default list, could you also add `eclipse-openj9/openj9` and `adoptium/infrastructure`. Thanks.

Reply:
Added :smile: